docs/operational_guide/topology.md

```diff
@@ -0,0 +1,173 @@
+# Topology
+
+## Overview
```

Comment:
could you add lifecycle diagrams for state transitions along with arrows for who triggers any transition - e.g. an operator adds a shard to a node when it's being added in the `Initializing` state, the node transitions to `Available` once it's bootstrapped, and it goes to `Leaving` when an operator marks it for removal, and its removed from the placement entirely by the other replica during a join goes from `Initializing` to `Available.